### PR TITLE
Fix EVEREST name to uppercase

### PR DESCRIPTION
**Pull request by roliveira**
**Issue**
After the open source of EVEREST it was agreed to use a trademark:

> EVEREST™ should have a trademark symbol the first time it appears in a publication.

but also to refer to it as an all caps, instead of the originally named EVEReST (Ensemble based Value Evaluation for Reservoir Simulation Tool), or other variations of the name.

Resolves #my_issue

**Approach**
Changed the ocurrances of `Everest` to `EVEREST` in docs, docstrings and a few comments. 

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

File: docs/everest/conf.py [hunk 1]
```diff
@@ -173,15 +173,15 @@
 # (source start file, target name, title,
 #  author, documentclass [howto, manual, or own class]).
 latex_documents = [
-    (master_doc, "Everest.tex", "Everest Documentation", "Equinor \\& TNO", "manual"),
+    (master_doc, "EVEREST.tex", "EVEREST Documentation", "Equinor \\& TNO", "manual"),
```
Comment (andreas-el) [inline docs/everest/conf.py]: Is the intent that we want uppercase names? Can be done, but this does not immediately sit right with me.

File: docs/everest/conf.py [hunk 2]
```diff
@@ -173,15 +173,15 @@
...
 ]
 
 
 # -- Options for manual page output ------------------------------------------
 
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
-man_pages = [(master_doc, "everest", "Everest Documentation", [author], 1)]
+man_pages = [(master_doc, "everest", "EVEREST Documentation", [author], 1)]
```
Comment (andreas-el) [inline docs/everest/conf.py:184]: This somewhat contradicts the previous entry, as the filename now is lowercase?

File: docs/everest/conf.py [hunk 3]
```diff
@@ -192,10 +192,10 @@
 texinfo_documents = [
     (
         master_doc,
-        "Everest",
-        "Everest Documentation",
+        "EVEREST",
```
Comment (andreas-el) [inline docs/everest/conf.py]: And uppercase file again. I think we should try to pick one of these styles, and stick with that.

File: tests/everest/test_everest_client.py [hunk 4]
```diff
@@ -242,9 +242,9 @@ def everserver_is_running():
     first = client_event_lists[0]
     assert all(first == other for other in client_event_lists[1:])
 
-    first_everevents = [e.event_type for e in first if "Everest" in e.event_type]
+    first_everevents = [e.event_type for e in first if "EVEREST" in e.event_type]
     server_everevents = [
-        e.event_type for e in server_events_list if "Everest" in e.event_type
```
Comment (andreas-el) [inline tests/everest/test_everest_client.py:247]: Could you add an assertion to this test also? Verifying that the two lists are not empty?

File: src/ert/run_models/everest_run_model.py [hunk 5]
```diff
@@ -588,7 +588,7 @@ def start_simulations_thread(
         error_messages: MutableSequence[str] = []
         traceback_str: str | None = None
         try:
-            logger.debug("Starting Everest simulations thread")
+            logger.debug("Starting EVEREST simulations thread")
```
Comment (andreas-el) [inline src/ert/run_models/everest_run_model.py:617]: We'd have to verify that any kusto queries handle case insensitive naming, otherwise the grafana dashboards could show wrong statistics after this.

Comment (andreas-el) [reply src/ert/run_models/everest_run_model.py:617]: After checking I found that the `contains` operator in `kusto` is case insensitive. 🎉

File: tests/everest/test_everest_client.py (hunk 4 above)
Comment (roliveira) [reply tests/everest/test_everest_client.py:247]: Great catch @andreas-el! This was actually wrong, but the test still passed (assert first_everevents == server_everevents --> 0 == 0)

I've added your suggestion to check if both event lists are not empty but also changed the test to check for the class instance itself instead of a string casting.
